### PR TITLE
Fix test failures

### PR DESCRIPTION
**Pull request by ellensi**
#### 1. Fix test failures related to RecordableCommand
* Force disable analytics tracking under recorded tests.
* Yet still, never track a `CreateNodeEvent` if by any chance the `Node` is null, by adding null checking inside the recently introduced methods `CreateNodeCommand.TrackAnalytics()` and `NodeModelExtensions.GetOriginalName()`.
* Capture better stack trace for exceptions that are thrown inside a recorded test.

##### Stack trace in NUnit before change:
image

##### After change:
image

#### 2. Fix backward compatibility test failures related to LibraryServices
* We now have two constructors that accept 2 arguments and 3 arguments, to ensure binary compatibility.

### Declarations

- [x] The code base is in a better state after this PR
- [x] All tests pass using the self-service CI.

### Reviewers

@Benglin @ke-yu

### FYI

@aparajit-pratap

### CODE REVIEW

**Comment by sharadkjaiswal** [comment]
Lgtm

**Comment by mjkkirschner** [comment]
are there any other changes required here before a merge?

File: src/DynamoCore/Graph/Nodes/NodeModelExtensions.cs [hunk 1]
```diff
@@ -99,6 +99,9 @@ internal static List<IGraphicItem> GeneratedGraphicItems(this NodeModel node, En
 
         internal static string GetOriginalName(this NodeModel node)
         {
+            if (node == null)
+                return string.Empty;
+
```
Comment (Benglin) [inline src/DynamoCore/Graph/Nodes/NodeModelExtensions.cs]: Refer to [coding standard](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards) for a recommendation on single-line statements.

File: src/DynamoCore/Models/RecordableCommands.cs [hunk 2]
```diff
@@ -799,6 +799,8 @@ protected override void SerializeCore(XmlElement element)
 
             internal override void TrackAnalytics()
             {
+                if (Node == null) return;
+
                 Dynamo.Logging.Analytics.TrackEvent(
                     Logging.Actions.Create,
                     Logging.Categories.NodeOperations,
```
Comment (Benglin) [inline src/DynamoCore/Models/RecordableCommands.cs]: It's better to do this:

```csharp
Dynamo.Logging.Analytics.TrackEvent(
    Logging.Actions.Create,
    Logging.Categories.NodeOperations,
    Node == null ? Name : Node.GetOriginalName());
```

**Comment by Benglin** [comment]
Yep there are changes required prior to the merge, once those are in, LTGM!

**Comment by ellensi** [comment]
The required changes are in. Thanks @Benglin and Sharad for reviewing!

**Comment by mjkkirschner** [comment]
@ellensi is this merged to RC1.1 yet? I think there are still failures on the build there.

**Pull request merged**